formula/src/main/java/com/instacart/formula/internal/FormulaManagerImpl.kt

```diff
@@ -245,6 +245,40 @@ internal class FormulaManagerImpl<Input, State, Output>(
         return manager.run(input).output
     }
 
+    fun <ChildInput, ChildOutput> child(
+        key: Any,
+        formula: IFormula<ChildInput, ChildOutput>,
+        input: ChildInput,
+        onError: (Throwable) -> Unit,
+    ): ChildOutput? {
+        val childrenManager = getOrInitChildrenManager()
+        val manager = childrenManager.findOrInitChild(key, formula, input)
+
+        // If termination happens while running, we might still be initializing child formulas. To
+        // ensure correct behavior, we mark each requested child manager as terminated to avoid
+        // starting new actions.
+        if (isTerminated()) {
+            manager.markAsTerminated()
+            manager.performTerminationSideEffects()
```

Comment:
I'm concerned that performTerminationSideEffects will get executed more than necessary. I've added some code to poll the transition queue on termination to avoid duplicate effect execution, but calling this more than once will have undesired effects such as an extra inspector::onFormulaFInished call. 

@Laimiux can you elaborate a little on the expectations of when performTerminationSideEffects is going to be called?